### PR TITLE
Update pop from 5.0.4 to 5.0.10

### PR DESCRIPTION
**Pull request by lotia**
Created with `brew bump-cask-pr`.

### CODE REVIEW

**Review by carlocab — DISMISSED** [review]

**Comment by cho-m** [comment]
Looks like Pop has changed behavior of website that we were using for `livecheck`.

I couldn't find an alternative, but will wait to see if anyone else finds one.
If nothing is available, we will probably need to just skip livecheck.

From basic scan through network capture, I only see some version information being passed over WebSocket (wss://).

**Comment by lotia** [comment]
Thanks for the comments and for looking into it @cho-m I had a look at the available strategies and none of them look like they will work for the updated download page.

My *clearly limited* understanding of the PageMatch and Git strategies is that they both depend on some parseable content being returned at the specified livecheck URL. There isn't a page that returns version parseable data that can be used to determine a version. I've updated the PR having skipped the livecheck.

**Comment by carlocab** [comment]
@samford or @nandahkrishna may have ideas here. (Sorry, both. Had to call in the experts...)

**Comment by nandahkrishna** [comment]
Seems like we might be able to check https://pop.com/download/mac? There's some text there (`Version 5.0.10`) which we can match.

**Comment by samford** [comment]
> Seems like we might be able to check https://pop.com/download/mac? There's some text there (`Version 5.0.10`) which we can match.

Unfortunately it's rendered using JavaScript (i.e., the text isn't in the HTML). I'm digging around to see if I can find where the version information comes from.

**Comment by cho-m** [comment]
When I last checked, the version was sent over WebSockets
If I block firebase.io, the version is not able to display.

File: Casks/pop.rb [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
-  sha256 "6e78f5d3b475ec6721de8e2024194a97a40abc5a65a264f60a2395d9b1bb88a2"
+  version "5.0.10"
+  sha256 "e94ddddb5839f349bee3757c2adccccc5f6bf2ba8da9f555b88d92bf40b2fec4"
 
   url "https://download.pop.com/desktop-app/darwin/#{version}/Pop.dmg"
   name "Pop"
   desc "Remote pair programming"
   homepage "https://pop.com/"
 
   livecheck do
```
Comment (samford) [inline Casks/pop.rb:10]:
```suggestion
  # The download page (https://pop.com/download/mac) is rendered using
  # JavaScript and the version information is sent over a WebSocket connection.
  # As such, we can't check it and there don't appear to be any alternatives.
  livecheck do
```

File: Casks/pop.rb [hunk 2]
```diff
@@ -1,16 +1,14 @@
...
-    url "https://download.pop.com/"
-    strategy :page_match
-    regex(%r{desktop-app/darwin/(\d+(\.\d+)*)/Pop\.dmg})
+    skip "File listing unavailable"
```
Comment (samford) [inline Casks/pop.rb:11]:
```suggestion
    skip "Version information not available in a checkable format"
```

**Review by samford — CHANGES_REQUESTED** [review]
I didn't see any available alternatives that we can check, so I'm fine with skipping this. I've added some suggestions that explain the situation in a preceding comment and clarify the `skip` message.

**Review by vitorgalvao — APPROVED** [review]

File: Casks/pop.rb [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
   desc "Remote pair programming"
   homepage "https://pop.com/"
 
   livecheck do
-    url "https://download.pop.com/"
-    strategy :page_match
-    regex(%r{desktop-app/darwin/(\d+(\.\d+)*)/Pop\.dmg})
+    # The download page (https://pop.com/download/mac) is rendered using
+    # JavaScript and the version information is sent over a WebSocket connection.
+    # As such, we can't check it and there don't appear to be any alternatives.
```
Comment (samford) [inline Casks/pop.rb:13]: @vitorgalvao  For what it's worth, we originally standardized on putting the explanatory comment before the `livecheck` block in homebrew/livecheck and this has remained true in homebrew/core. I would prefer to continue the existing approach, so this information is always in the same place.

We do have some brief inline comments within `livecheck` blocks in a small number of instances (mostly `strategy` blocks) but the general explanatory comment always comes before the `livecheck` block.

Unfortunately I wasn't fast enough to catch this before it auto-merged.

Comment (vitorgalvao) [reply Casks/pop.rb:13]: https://github.com/Homebrew/homebrew-cask/pull/106034

**Comment by lotia** [comment]
Thanks for providing super valuable demonstration of how to do it right. Much appreciated.

**Pull request merged**